### PR TITLE
Add type declaration to `MigrationCollector::reset()`

### PR DESCRIPTION
**Pull request by asuri0n**
Added the explicit @return type in `MigrationCollector::reset()` to suspend the deprecation info msg

image

### CODE REVIEW

File: Collector/MigrationsCollector.php [hunk 1]
```diff
@@ -82,6 +82,9 @@ public function getData()
         return $this->data;
     }
 
+    /**
+     * {@inheritDoc}
```
Comment (derrabus) [inline Collector/MigrationsCollector.php]: That's not enough to silence the deprecation emitted by Symfony's error handler. You need to document the return type explicitly.

```suggestion
     * @return void
```

**Review by SenseException — CHANGES_REQUESTED** [review]
Please use a meaningful commit message as described in our [contribution guide](https://www.doctrine-project.org/contribute/index.html#crafting-meaningful-commit-messages).

**Comment by asuri0n** [comment]
@SenseException I don't know how to edit my previous commit messages

**Comment by stof** [comment]
you need to amend your git history locally and to force-push your branch after that

**Comment by SenseException** [comment]
@asuri0n You can find infos on how to change the last commit here: https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History#_git_amend and do a force-push afterwards.

**Comment by asuri0n** [comment]
well ... I think I screwed up the git history.... I will make a new merge request with clean commit message.

**Comment by greg0ire** [comment]
> I will make a new merge request with clean commit message.

You don't need to, you can just force push once the history is good.

What you could do is

```
git reset 9432bc37ab3bc4f074425c669d8a63cc388b17df
git add --patch
git commit --verbose
git push --force-with-lease
```

**Comment by greg0ire** [comment]
Cc @mpdude, regarding https://github.com/doctrine/doctrine-website/pull/473#discussion_r1129076152 :)

**Comment by greg0ire** [comment]
@asuri0n there are 9 commits now. Have you tried what I recommended above?

**Comment by mpdude** [comment]
If we‘d ask contributors to provide meaningful summaries in the opening comment, it would help everybody coming here for a review, and less experienced Git users could more easily change the text than with Git history rewrites and force-pushes.

Then, in the end, squash-merge and use the initial comment as the squash merge message. That also makes meaningless commits in the course of the PR („Fix CS“) irrelevant.

Would have saved us all some headache here, non? But that’s just my $.02, off topic here so I’ll just shut up 😉

**Comment by greg0ire** [comment]
No no, you're right, I think it's just we didn't think of that! @asuri0n , can you please edit the body of the pull request, as suggested?

**Comment by asuri0n** [comment]
It's a pretty simple pull request, I've updated the body, is it enough ?

**Comment by greg0ire** [comment]
> to suspend the deprecation info msg

It would be nice to quote the message. Also, it would be nice to know why you fixed `reset()` but not `getData()`. Is there no message for that one?

I think it's a deprecation coming from Symfony, right? And I think maybe you considered using `@{inheritDoc}`, but that is not enough for some reason I don't remember?

**Comment by derrabus** [comment]
> And I think maybe you considered using `@{inheritDoc}`, but that is not enough for some reason I don't remember?

https://github.com/doctrine/DoctrineMigrationsBundle/pull/491#discussion_r1135220070

**Comment by greg0ire** [comment]
Oh right :facepalm:

**Review by derrabus — APPROVED** [review]

**Comment by derrabus** [comment]
Thank you @asuri0n!

**Pull request merged**